### ISSUE TITLE
Pubmed query not working

### ISSUE THREAD

**Issue by corrad82-4s** [DSpace/DSpace#8327]
**Describe the bug**
When a new submission is started from a query against Pubmed external system, the query does not return any result. 

**To Reproduce**
1. login to dspace7 and go to MyDSpace
2. start a new Publication submission from external source
3. enter a query string and select “Pubmed” as external source

Call to backend performing Pubmed query will fail with 500 error (https://api7.dspace.org/server/api/integration/externalsources/pubmed/entries?sort=id,ASC&page=0&size=10&query=covid19)

**Expected behavior**
A query returning results.

Running REST locally to investigate the bug, we found out that the first ever query on Pubmed performed when REST backend is up and running returns results, whereas next queries (second, third,...) ends up with no results and 500 error.

Our proposal is, instead of just fixing this bug, operate a refactoring to Pubmed integration and align it to the ones recently merged in DSpace (e.g. CrossRef), then add some tests on Pubmed LiveImport connector refactored. 
We (4Science) can take in charge this activity. Our estimation for refactoring and test addition is 8/12 hours of work

**Comment by tdonohue** [comment]
@corrad82-4s : Can this be fixed immediately?  It seems like we should minimally fix the Pubmed query in 7.3...even if it means delaying the larger refactoring.

In any case, I'll assign this back to you and move to the 7.3 board.  My preference would be to find a "quick fix" for 7.3, even if that means that the larger refactor needs to wait for 7.4.  Either way, I'd gladly help review this once it is ready.  Thanks!

**Comment by Micheleboychuk** [comment]
Hi @tdonohue, I did an initial analysis of the problem and found that the problem lies in the pabmed bees, so it is due to an external problem. Let me explain in more detail, for the same request we make to pubmed, pubmed once answers with 200 another time it might answer with 500, to simulate this just do refreshe (F5) following the behaviour described by @corrad82-4s.

A possible solution: when on the rest side we receive a 500 response from pubmed, we can retry the request for a finite number of times (perhaps configurable in dspace.cfg). otherwise we relaunch the eception.

**Comment by tdonohue** [comment]
@Micheleboychuk : I'm ok with a simple retry on the backend if you feel that'd solve the issue here.  I agree it should be a finite number of times...I'm OK with it being configurable, but that's not required in my opinion.   Overall, I'd just like to find a very quick solution as otherwise we'll need to leave this bug in the 7.3 release.

**Comment by tdonohue** [comment]
@Micheleboychuk : A quick update... I just realized that Sherpa has a similar retry option, so you might be able to build something similar for PubMed.  See https://github.com/DSpace/DSpace/blob/main/dspace/config/spring/api/external-services.xml#L28-L32

**Comment by bram-atmire** [comment]
We are still seeing this problem on 7.4 instances. Is anyone else seeing the same?
I'm wondering if the previous solution, that added additional retry requests, might not be effective if the original root cause is rate limiting/throttling on the end of Pubmed?

Background: https://www.ncbi.nlm.nih.gov/books/NBK25497/

TL;DR => if they catch you with more than 3 requests per second, the IP of your server might get flagged.
You can also request an API key (optionally), which gets you more quota.

So maybe we have to go into the direction of adding that api key as an extra config var that people can add in the config, and that, if an API key is configured, the requests will go out with the key?

That is of course if I'm on the right track with the hypothesis that rate limiting/throttling is at the root of this issue, which we have not yet been able to prove but we're looking into it further.

**Comment by tdonohue** [comment]
@bram-atmire : Please create a new ticket for your suggestions... those are out of scope for this original closed ticket. It sounds like you are requesting an enhancement to add an API key to these requests. That's a reasonable request, but we would need to find a *new* volunteer to work on that, and not try to add it to an already closed ticket (as this ticket involved work previously paid for).  Please always open a new ticket for a new request/bug & feel free to link back to older related tickets.
